### PR TITLE
`azurerm_eventhub_authorization_rule` - Fix intermittent 404 errors

### LINKED ISSUE #6983: Intermittent issues when creating azurerm_eventhub_authorization_rule
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform version: v0.12.21
azurerm version: 2.9.0

### Affected Resource(s)

* `azurerm_eventhub_authorization_rule`

### Terraform Configuration Files

```hcl
provider azurerm {
  version = "2.9.0"
  features {}
}

locals {
  # Namespace
  namespace_count                     = 1 # Based on some other logic, but 1 for example sake
  event_hub_namespace_name_with_count = join("", concat(azurerm_eventhub_namespace.example.*.name))

  # Diagnostic logs
  diagnostic_logs_count                     = 1 # Based on some other logic, but 1 for example sake
  diagnostic_logs_event_hub_name_with_count = join("", concat(azurerm_eventhub.diagnostic_logs.*.name))

  # Application logs
  application_logs_count                     = 1 # Based on some other logic, but 1 for example sake
  application_logs_event_hub_name_with_count = join("", concat(azurerm_eventhub.application_logs.*.name))
}

resource "azurerm_resource_group" "example" {
  name     = "example-rg"
  location = "centralus"
}

resource "azurerm_eventhub_namespace" "example" {
  count = local.namespace_count

  name                = "example-ehn"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1
}

resource "azurerm_eventhub" "diagnostic_logs" {
  count = local.diagnostic_logs_count

  name                = "diagnostic-logs"
  namespace_name      = local.event_hub_namespace_name_with_count
  resource_group_name = azurerm_resource_group.example.name
  partition_count     = 32
  message_retention   = 7
}

resource "azurerm_eventhub_authorization_rule" "diagnostic_logs" {
  name                = "auth"
  namespace_name      = local.event_hub_namespace_name_with_count
  eventhub_name       = local.diagnostic_logs_event_hub_name_with_count
  resource_group_name = azurerm_resource_group.example.name
  listen              = true
  send                = true
  manage              = false
}

resource "azurerm_eventhub" "application_logs" {
  count = local.application_logs_count

  name                = "application-logs"
  namespace_name      = local.event_hub_namespace_name_with_count
  resource_group_name = azurerm_resource_group.example.name
  partition_count     = 32
  message_retention   = 7
}

resource "azurerm_eventhub_authorization_rule" "application_logs" {
  name                = "auth"
  namespace_name      = local.event_hub_namespace_name_with_count
  eventhub_name       = local.application_logs_event_hub_name_with_count
  resource_group_name = azurerm_resource_group.example.name
  listen              = true
  send                = true
  manage              = false
}
```

### Debug Output

```
eventhub.EventHubsClient#GetAuthorizationRule: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="NotFound" Message="The requested resource auth does not exist. CorrelationId: c70f82e7-8209-72ff-f08c-3c14d01aaccf"

  on main.tf line 47, in resource "azurerm_eventhub_authorization_rule" "diagnostic_logs":
  47: resource "azurerm_eventhub_authorization_rule" "diagnostic_logs" {
```

The next `terraform apply` will fail because the rule already exists and it says it needs to be imported into state.

### Panic Output

None

### Expected Behavior

The Event Hub authorization rule should have been created without issue.

### Actual Behavior

The Event Hub authorization rule was created (verified in Azure portal) but Terraform had the above error message.

### Steps to Reproduce

This issue seems to be Intermittent. After around ten times of running `terraform apply` and `terraform destroy`, I was eventually able to get the above error message after a `terraform apply`.

### Important Factoids

None

### References

The above error message is similar to the one mentioned in [this](https://github.com/terraform-providers/terraform-provider-azurerm/issues/4893) issue. The difference is that I am only creating one rule for each event hub where the referenced issue was creating multiple rules for a single event hub. I'm guessing this is potentially occurring [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v2.9.0/azurerm/internal/services/eventhub/eventhub_authorization_rule_resource.go#L109) where the rule is read back after creation? The rule may not be getting fully created before the read occurs?

### LINKED ISSUE #7132: Error making Read request on Azure EventHub Authorization Rule
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `Terraform v0.12.26` for mac x64
* `provider.azurerm v2.12.0`

### Terraform Configuration Files

```
provider "azurerm" {
  version = ">= 2.10.0"
  features {}
}

resource "azurerm_resource_group" "gc" {
        name = "gcResourceGroup"
        location = "westus"
}

resource "azurerm_eventhub_namespace" "gc" {
  name                = "gcEventHubNamespace"
  location            = azurerm_resource_group.gc.location
  resource_group_name = azurerm_resource_group.gc.name
  sku                 = "Basic"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub" "gc" {
  name                = "gcEventHub"
  namespace_name      = azurerm_eventhub_namespace.gc.name
  resource_group_name = azurerm_resource_group.gc.name
  partition_count     = 2
  message_retention   = 1
}

resource "azurerm_eventhub_namespace_authorization_rule" "gc" {
  name                = "gcEventHubNamespaceAuthRule"
  namespace_name      = azurerm_eventhub_namespace.gc.name
  resource_group_name = azurerm_resource_group.gc.name

  listen = true
  send   = false
  manage = false
}
```

### Debug Output

```
azurerm_eventhub_namespace_authorization_rule.gc: Refreshing state... [id=/subscriptions/xx-xx/resourceGroups/gcResourceGroup/providers/Microsoft.EventHub/namespaces/gcEventHubNamespace/authorizationRules/gcEventHubNamespaceAuthRule]
..
Error: Error making Read request on Azure EventHub Authorization Rule : eventhub.NamespacesClient#GetAuthorizationRule: Invalid input: autorest/validation: validation failed: parameter=authorizationRuleName constraint=MinLength value="" details: value length must be greater than or equal to 1
```

### Expected Behavior
`terraform apply` should complete without errors.
Resource group `gcResourceGroup` and eventhub `gcEventHub` within eventhub namespace `gcEventHubNamespace` should be deployed and Auth Rule created on `gcEventHubNamespace`.

### Actual Behavior

```
Resource group and all resources deployed but command ends with error.
```
### Steps to Reproduce

Copy simple infra code and perform `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by nickmhankins**
* Add retry for Event Hub Authorization Rule create to prevent 404 error, since adding Locks didn't seem to do the trick (Fixes #6983 #4893)
* Fix 404 error when deleting rules
* Update case for Namespace Authorization Rules API path to account for Azure Go SDK update (#6725) (Fixes #7132)

### CODE REVIEW

**Comment by nickmhankins** [comment]
Looks like this failed for other reasons:

```
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $GOPATH/bin v1.24.0

golangci/golangci-lint info checking GitHub for tag 'v1.24.0'

golangci/golangci-lint info found version: 1.24.0 for v1.24.0/linux/amd64

GNUmakefile:14: recipe for target 'tools' failed

make: *** [tools] Error 1

The command "make tools" failed and exited with 2 during .
```

File: azurerm/internal/services/eventhub/eventhub_authorization_rule_resource.go [hunk 1]
```diff
@@ -102,22 +102,24 @@ func resourceArmEventHubAuthorizationRuleCreateUpdate(d *schema.ResourceData, me
...
+			return resource.NonRetryableError(fmt.Errorf("Error creating Authorization Rule %q (event Hub %q / Namespace %q / Resource Group %q): %+v", name, eventHubName, namespaceName, resourceGroup, err))
+		}
 
-	read, err := client.GetAuthorizationRule(ctx, resourceGroup, namespaceName, eventHubName, name)
-	if err != nil {
-		return err
-	}
+		read, err := client.GetAuthorizationRule(ctx, resourceGroup, namespaceName, eventHubName, name)
+		if err != nil {
+			return resource.RetryableError(fmt.Errorf("Expected instance of the authorization rule %q (event Hub %q / Namespace %q / Resource Group %q) to be created but was in non existent state, retrying", name, eventHubName, namespaceName, resourceGroup))
```
Comment (mbfrahry) [inline azurerm/internal/services/eventhub/eventhub_authorization_rule_resource.go:112]: What're your thoughts on retrying only when we error on a 404? Make it a little more specific to the error we are trying to solve

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @nickmhankins, this looks good! I just think we can be a little more specific on the RetryableError piece

File: azurerm/internal/services/eventhub/eventhub_authorization_rule_resource.go (hunk 1 above)
Comment (nickmhankins) [reply azurerm/internal/services/eventhub/eventhub_authorization_rule_resource.go:112]: That's a great idea, i'll work on that

Comment (nickmhankins) [reply azurerm/internal/services/eventhub/eventhub_authorization_rule_resource.go]: @mbfrahry Requested change has been made

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**